test/src/api/responder04Test.php

```diff
@@ -0,0 +1,916 @@
 any number of dice which add up exactly to its value."}}, "gameActionLog": [{"timestamp": 1675914367, "message": "Game created by responder003", "player": ""}], "maxWins": 3, "playerDataArray": [{"prevOptValueArray": [], "optRequestArray": [], "playerColor": "#dd99dd", "roundScore": null, "waitingOnAction": true, "playerId": 3, "outOfPlayDieArray": [], "button": {"artFilename": "boot2dahead.png", "recipe": "d(1) Gz(Z) !p(V) !n(Y,Y) oF(C)", "name": "Boot2daHead", "originalRecipe": "d(1) Gz(Z) !p(V) !n(Y,Y) oF(C)"}, "capturedDieArray": [], "isChatPrivate": false, "isOnVacation": false, "activeDieArray": [{"description": "Stealth 1-sided die", "skills": ["Stealth"], "recipe": "d(1)", "value": null, "properties": [], "sides": 1}, {"description": "Rage Speed Z Swing Die", "skills": ["Rage", "Speed"], "recipe": "Gz(Z)", "value": null, "properties": [], "sides": null}, {"description": "Poison Turbo V Swing Die", "skills": ["Poison", "Turbo"], "recipe": "p(V)!", "value": null, "properties": [], "sides": null}, {"description": "Null Turbo Twin Y Swing Die", "skills": ["Null", "Turbo"], "recipe": "n(Y,Y)!", "value": null, "properties": ["Twin"], "sides": null}, {"description": "Ornery Fire Wildcard die", "skills": ["Ornery", "Fire"], "recipe": "oF(C)", "value": null, "properties": [], "sides": 20}], "swingRequestArray": {"Y": [1, 20], "Z": [4, 30], "V": [6, 12]}, "hasDismissedGame": false, "gameScoreArray": {"L": 0, "W": 0, "D": 0}, "sideScore": null, "canStillWin": null, "turboSizeArray": [], "playerName": "responder003", "prevSwingValueArray": [], "lastActionTime": 1675914367}, {"prevOptValueArray": [], "optRequestArray": [], "playerColor": "#ddffdd", "roundScore": null, "waitingOnAction": true, "playerId": 5, "outOfPlayDieArray": [], "button": {"artFilename": "envy.png", "recipe": "D(4) D(6) D(10) D(12) D(X)", "name": "Envy", "originalRecipe": "D(4) D(6) D(10) D(12) D(X)"}, "capturedDieArray": [], "isChatPrivate": false, "isOnVacation": false, "activeDieArray": [{"description": "Doppelganger 4-sided die", "skills": ["Doppelganger"], "recipe": "D(4)", "value": null, "properties": [], "sides": 4}, {"description": "Doppelganger 6-sided die", "skills": ["Doppelganger"], "recipe": "D(6)", "value": null, "properties": [], "sides": 6}, {"description": "Doppelganger 10-sided die", "skills": ["Doppelganger"], "recipe": "D(10)", "value": null, "properties": [], "sides": 10}, {"description": "Doppelganger 12-sided die", "skills": ["Doppelganger"], "recipe": "D(12)", "value": null, "properties": [], "sides": 12}, {"description": "Doppelganger X Swing Die", "skills": ["Doppelganger"], "recipe": "D(X)", "value": null, "properties": [], "sides": null}], "swingRequestArray": {"X": [4, 20]}, "hasDismissedGame": false, "gameScoreArray": {"L": 0, "W": 0, "D": 0}, "sideScore": null, "canStillWin": null, "turboSizeArray": [], "playerName": "responder004", "prevSwingValueArray": [], "lastActionTime": 1675914367}], "gameChatLogCount": 0, "validAttackTypeArray": []}', TRUE));
```

Comment:
TODO: assuming we keep this test, we'll need to clean this up before submitting.  (I can do it, just put a reminder so we don't forget and merge this, because the blob of text will make bookkeeping harder.)